docs/contributing/coding-style.md

```diff
@@ -965,3 +965,98 @@ We do not suggest to modify the generated code other than the `tests` variable,
         type want struct {
             // generated test code
     ```
+### Mock
+
+In Vald, we use a lot of external library, there are a lot of dependencies between libraries.
+
+As a result, the complexity of the test has increased, and it has become more difficult to determine whether or not to mock dependent objects.
+
+#### Condition
+
+When a dependent object has the following feature, you can decide to mock the dependent.
+
+- Incomplete implementation
+- IO
+  - Network access, disk operation, etc.
+- Hardware dependent
+  - CPU, memory usage, disk IO, etc.
+- Difficult to create error of dependent object (when we will write error test case)
+- Difficult to initialize
+  - Random number and time, file IO initialization, environment dependent, etc.
+- Test result may change in runtime
+  - Cases where the implementation and test code are not changed but the test result changes
+  - System call inside implementation
+
+#### Implementation
+
+1. Basic mock example.
+
+    For example, we decided to mock the following implementation `Encoder`.
+
+    ```go
+    package json
+
+    type Encoder struct {
```

Comment:
is it an interface?

Reply:
thank you. I made a misstake. so I will fix it soon.